python/lsst/ts/aos/utils/m1m3/m1m3_aav.py

```diff
@@ -0,0 +1,855 @@
+            """Retrieve axis data.
+
+            Parameters
+            ----------
+            axis : `str`
+                Axis to retrieve. Either elevation or azimuth.
+
+            Returns
+            -------
+            ret : pd.DataFrame
+                DataFrame with axis data. demand and actual Accelerations are
+                calculated as difference in speed from the last datapoint,
+                divided by interval between the two measurements.
+            """
+            logging.info(f"Querying EFD for {axis} data")
+            ret = await self.client.select_time_series(
+                f"lsst.sal.MTMount.{axis}",
+                [
+                    "timestamp",
+                    "demandPosition",
+                    "actualPosition",
+                    "demandVelocity",
+                    "actualVelocity",
+                ],
+                start,
+                end,
+            )
+
+            ret.set_index(
+                pd.DatetimeIndex(
+                    Time(Time(ret["timestamp"], format="unix_tai"), scale="utc").isot
+                ),
+                inplace=True,
+            )
+            ret["timediff"] = ret["timestamp"].diff()
+
+            # calculate derivatives - acceleration
+            den = ret["timediff"]
+            ret["demandAcceleration"] = ret["demandVelocity"].diff().div(den, axis=0)
+            ret["actualAcceleration"] = ret["actualVelocity"].diff().div(den, axis=0)
+            logging.info(
+                f"Retrieved {axis} {len(ret.index)} rows. "
+                f"Ranges: {ret.demandPosition.min():+.2f} .. "
+                f"{ret.demandPosition.max():+.2f}"
+                f", {ret.actualPosition.min():+.2f} .. "
+                f"{ret.actualPosition.max():+.2f}"
+            )
+            return ret
+
+        self.elevations = await find_axis("elevation")
+        self.azimuths = await find_axis("azimuth")
+
+    async def find_applicable_times(self) -> None:
+        """Find start and end times of any axis movents. Uses self.elevations
+        and self.azimuths data retrieved in find_az_el method.
+
+        Fills self.intervals with start and stop times usefull for fitting.
+        """
+        assert self.azimuths is not None
+        assert self.elevations is not None
+
+        ret = []
+
+        movements = pd.concat(
+            [
+                self.elevations[
+                    (abs(self.elevations.actualVelocity) > self.actual_velocity_limit)
+                    | (abs(self.elevations.demandVelocity) > self.demand_velocity_limit)
+                ],
+                self.azimuths[
+                    (abs(self.azimuths.actualVelocity) > self.actual_velocity_limit)
+                    | (abs(self.azimuths.demandVelocity) > self.demand_velocity_limit)
+                ],
+            ]
+        )
+
+        movements.sort_index(inplace=True)
+        movements["timediff"] = movements.index.to_series().diff()
```

Comment:
Isn't this already available as movements["timediff"]? In line 258 in the previous function you define the timediff, right?

Reply:
But the one from 258 is either for azimuth or elevation. It's correct to compute it again, as this timediff is used to trigger where new slew starts (timediff > 1 sec from previous non-zero movement).